posthog/clickhouse/cancel.py

```diff
@@ -2,13 +2,44 @@
 
 from posthog.api.services.query import logger
 from posthog.clickhouse.client import sync_execute
+from posthog.clickhouse.client.connection import default_client
 from posthog.settings import CLICKHOUSE_CLUSTER
 
 
 def cancel_query_on_cluster(team_id: int, client_query_id: str) -> None:
-    result = sync_execute(
-        f"KILL QUERY ON CLUSTER '{CLICKHOUSE_CLUSTER}' WHERE query_id LIKE %(client_query_id)s",
-        {"client_query_id": f"{team_id}_{client_query_id}%"},
-    )
+    initiator_host = None
+    failed_to_find_initiator_host = False
+    try:
+        result = sync_execute(
+            f"""
+            SELECT hostname()
+            FROM clusterAllReplicas(posthog, system.query_log)
+            WHERE query_id LIKE %(client_query_id)s AND (event_time > (now() - toIntervalDay(1)))
+            GROUP BY hostname(), query_id -- There can be multiple queries with the same client_query_id
+            HAVING (count() = 1)
+            SETTINGS max_execution_time = 5
+            """,
+            {"client_query_id": f"{team_id}_{client_query_id}%"},
+        )
+        initiator_host = result[0][0] if result else None
+    except Exception as e:
+        failed_to_find_initiator_host = True
+        logger.info("Failed to find initiator host for query %s: %s", client_query_id, e)
+
+    if not failed_to_find_initiator_host:
```

Comment:
i think initiator_host and failed_to_find_initiator_host are redundant, initiator_host is never not None if failed_to_find_initiator_host is True

Reply:
The thing here is that the query may not fail and the `initiator_host` can be none if the query does not return any results (for example because the query we are trying to kill already finished).

Reply:
Ok, got it, then i'd go for:
```
if initiated_host:
  # run desired kill
elif failed_to_find_initiator_host:
  # run fallback kill on cluster
```

Reply:
Done, in fact, I removed the `failed_to_find_initiator_host`, because I have found cases in which the hostname() may not be found because the query was not still in the `query_log` (if we run / cancel quick enough). So I prefer to fallback to a KILL ON CLUSTER in case that's the case.